sink-connector/src/main/java/com/altinity/clickhouse/sink/connector/converters/DebeziumConverter.java

```diff
@@ -151,29 +152,53 @@ public static String convert(Object value) {
             // for example you might truncate microseconds
             // to milliseconds(3) if .SSS is above .SSSSSS
             String[] date_formats = {
-                    "yyyy-MM-dd'T'HH:mm:ss'Z'",
-                    "yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'",
-                    "yyyy-MM-dd'T'HH:mm:ss.SSSSS'Z'",
-                    "yyyy-MM-dd'T'HH:mm:ss.SSSS'Z'",
-                    "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'",
-                    "yyyy-MM-dd'T'HH:mm:ss.SS'Z'",
-                    "yyyy-MM-dd'T'HH:mm:ss.S'Z'"
+                    "yyyy-MM-dd'T'HH:mm:ss.SSSSSSSXXX",
+                    "yyyy-MM-dd'T'HH:mm:ss.SSSSSSXXX",
+                    "yyyy-MM-dd'T'HH:mm:ss.SSSSSXXX",
+                    "yyyy-MM-dd'T'HH:mm:ss.SSSSXXX",
+                    "yyyy-MM-dd'T'HH:mm:ss.SSSXXX",
+                    "yyyy-MM-dd'T'HH:mm:ss.SSXXX",
+                    "yyyy-MM-dd'T'HH:mm:ss.SXXX",
+                    "yyyy-MM-dd'T'HH:mm:ssXXX",
+                    "yyyy-MM-dd'T'HH:mm:ss.SSSSSSSZ",
+                    "yyyy-MM-dd'T'HH:mm:ss.SSSSSSZ",
+                    "yyyy-MM-dd'T'HH:mm:ss.SSSSSZ",
+                    "yyyy-MM-dd'T'HH:mm:ss.SSSSZ",
+                    "yyyy-MM-dd'T'HH:mm:ss.SSSZ",
+                    "yyyy-MM-dd'T'HH:mm:ss.SSZ",
+                    "yyyy-MM-dd'T'HH:mm:ss.SZ",
+                    "yyyy-MM-dd'T'HH:mm:ssZ"
             };
 
             boolean parsingSuccesful = false;
             for (String formatString : date_formats) {
                 try {
                     DateTimeFormatter formatter = DateTimeFormatter.ofPattern(formatString);
-                    LocalDateTime zd = LocalDateTime.parse((String) value, formatter);
-                    result = zd.format(destFormatter);
-                    //result = removeTrailingZeros(result);
+                    parsedDT = ZonedDateTime.parse((String) value, formatter);
                     parsingSuccesful = true;
                     break;
                 } catch(Exception e) {
+                    if (e.getMessage().contains("Invalid value for YearOfEra")) {
```

Comment:
Doesn't look like a good idea to check for exception using the message(Message could change in the future), isnt there a specific exception if parsing fails?

Reply:
OK, i will check.

Reply:
I have rewritten a bit this part. It's possible to use `getCause()` method that inherited from `Throwable` to define if it's either a `DateTimeParseException` error or a `DateTimeException`. If `getCause()` returns `null` - it's a parsing error, otherwise not.